runtime/near-vm-runner/src/logic/mocks/mock_external.rs

```diff
@@ -77,6 +77,7 @@ pub struct MockedExternal {
     pub fake_trie: HashMap<Vec<u8>, Vec<u8>>,
     pub validators: HashMap<AccountId, Balance>,
     pub action_log: Vec<MockAction>,
+    pub code_hash: CryptoHash,
```

Comment:
Issue is that code expects to be able to read out the `CodeHash` even if the contract is not provided. Usually it would then be stored with the `Account` and read out as `0xAAA...AAAA`. This is then used as a cache look-up key... I originally had this code set up such that it would read something like this:

```
code.map(|c| *c.hash()).unwrap_or_default() 
```

thus obviating storage of this field altogether, but it turns out that there are tests out there that depend on being able to set arbitrary "valid" code hash with an option to keep contract itself as a `None`, so I had to stick with this approach where both are stored separately.